File: petclinic-java-gradle-react-tailwind-ts/src/main/java/dev/pollito/petclinic_java_gradle_react_tailwind_ts/owner/OwnerServiceImpl.java
package dev.pollito.petclinic_java_gradle_react_tailwind_ts.owner;

import dev.pollito.petclinic_java_gradle_react_tailwind_ts.events.BeforeDeleteOwner;
import dev.pollito.petclinic_java_gradle_react_tailwind_ts.util.CustomCollectors;
import dev.pollito.petclinic_java_gradle_react_tailwind_ts.util.NotFoundException;
import java.util.Map;
import lombok.RequiredArgsConstructor;
import org.springframework.context.ApplicationEventPublisher;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.domain.Sort;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class OwnerServiceImpl implements OwnerService {

    private final OwnerRepository ownerRepository;
    private final ApplicationEventPublisher publisher;

    @Override
    public Page<Owner> findAll(final String filter, final Pageable pageable) {
        if (filter != null) {
            Integer integerFilter = null;
            try {
                integerFilter = Integer.parseInt(filter);
            } catch (final NumberFormatException numberFormatException) {
            }
            return ownerRepository.findAllById(integerFilter, pageable);
        }
        return ownerRepository.findAll(pageable);
    }

    @Override
    public Owner get(final Integer id) {
        return ownerRepository.findById(id)
                .orElseThrow(NotFoundException::new);
    }

    @Override
    public Integer create(final Owner owner) {
        return ownerRepository.save(owner).getId();
    }

    @Override
    public void update(final Integer id, final Owner owner) {
        owner.setId(id);
        ownerRepository.findById(id).orElseThrow(NotFoundException::new);
        ownerRepository.save(owner);
    }

    @Override
    public void delete(final Integer id) {
        final Owner owner = ownerRepository.findById(id)
                .orElseThrow(NotFoundException::new);
        publisher.publishEvent(new BeforeDeleteOwner(id));
        ownerRepository.delete(owner);
    }

    @Override
    public Map<Integer, Integer> getOwnerValues() {
        return ownerRepository.findAll(Sort.by("id"))
                .stream()
                .collect(CustomCollectors.toSortedMap(Owner::getId, Owner::getId));
    }
}

File: petclinic-kt-gradle-thymeleaf-tailwind-webpack-htmx-ts/src/main/kotlin/dev/pollito/petclinic_kt_gradle_thymeleaf_tailwind_webpack_htmx_ts/owner/OwnerServiceImpl.kt
package dev.pollito.petclinic_kt_gradle_thymeleaf_tailwind_webpack_htmx_ts.owner

import dev.pollito.petclinic_kt_gradle_thymeleaf_tailwind_webpack_htmx_ts.events.BeforeDeleteOwner
import dev.pollito.petclinic_kt_gradle_thymeleaf_tailwind_webpack_htmx_ts.util.CustomCollectors
import dev.pollito.petclinic_kt_gradle_thymeleaf_tailwind_webpack_htmx_ts.util.NotFoundException
import org.springframework.context.ApplicationEventPublisher
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.domain.Sort
import org.springframework.stereotype.Service

@Service
class OwnerServiceImpl(
    private val ownerRepository: OwnerRepository,
    private val publisher: ApplicationEventPublisher,
) : OwnerService {

    override fun findAll(filter: String?, pageable: Pageable): Page<Owner> = if (filter != null) {
        ownerRepository.findAllById(filter.toIntOrNull(), pageable)
    } else {
        ownerRepository.findAll(pageable)
    }

    override fun `get`(id: Int): Owner = ownerRepository.findById(id)
        .orElseThrow { NotFoundException() }

    override fun create(owner: Owner): Int = ownerRepository.save(owner).id!!

    override fun update(id: Int, owner: Owner) {
        owner.id = id
        ownerRepository.findById(id).orElseThrow { NotFoundException() }
        ownerRepository.save(owner)
    }

    override fun delete(id: Int) {
        val owner = ownerRepository.findById(id)
            .orElseThrow { NotFoundException() }
        publisher.publishEvent(BeforeDeleteOwner(id))
        ownerRepository.delete(owner)
    }

    override fun getOwnerValues(): Map<Int, Int> = ownerRepository.findAll(Sort.by("id"))
        .stream()
        .collect(CustomCollectors.toSortedMap(Owner::id, Owner::id))
}
